### PR TITLE
Introduce lifecycle hooks

### PR DESCRIPTION
**Pull request by mixonic**
* didUpdatePost, which gets passed the postEditor
* willRender, before render after post changes are complete
* didRender, after render when DOM and post are in sync

TODO:

* [x] Docs

### CODE REVIEW

**Comment by mixonic** [comment]
@bantic this is good to go.

File: src/js/editor/editor.js [hunk 1]
```diff
@@ -319,6 +332,8 @@ class Editor {
       showForTag: 'a'
     }));
 
+    // Stub out a post editing session for initial render
+    this.run(() => {});
```
Comment (bantic) [inline src/js/editor/editor.js]: Why is the empty `run` there? It's not clear to me from that comment.

Comment (mixonic) [inline src/js/editor/editor.js]: Will elaborate, poor wording there.

**Comment by mixonic** [comment]
:tada:

File: src/js/editor/editor.js [hunk 2]
```diff
@@ -319,6 +332,9 @@ class Editor {
       showForTag: 'a'
     }));
 
+    // A call to `run` will trigger the didUpdatePostCallbacks hooks with a
+    // postEditor.
+    this.run(() => {});
```
Comment (bantic) [inline src/js/editor/editor.js]: Maybe this should be `this.runCallbacks()` or something? This looks like the sort of no-op-y code that will get deleted by someone in the future.

**Pull request merged**